### PR TITLE
Fix notebooks instances recreating for scopes and CMEK

### PR DESCRIPTION
**Pull request by bcreddy-gcp**
- https://github.com/hashicorp/terraform-provider-google/issues/9164
- https://github.com/hashicorp/terraform-provider-google/issues/16587
- https://github.com/hashicorp/terraform-provider-google/issues/15274

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
The recent error with the acceptance tests are due to [this code being generated with empty strings](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9915-old..auto-pr-9915#diff-ff16b58fdbc846c15cf0e11be244458315ef81d5211f5ac9ed283bba4c6435ae) - the solution is updating [the variables used when templating out the acceptance test](https://github.com/GoogleCloudPlatform/magic-modules/blob/45d47286e2db29c5eb7b3982be24e57d50648dd2/mmv1/products/notebooks/Instance.yaml#L78-L79), e.g.:  

```diff
    vars:
      instance_name: 'notebooks-instance'
+      key_name: 'my-key'
+      keyring_name: 'my-keyring'
```

File: mmv1/templates/terraform/constants/notebooks_instance.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+var NotebooksInstanceProvidedScopes = []string{
+	"https://www.googleapis.com/auth/cloud-platform",
+	"https://www.googleapis.com/auth/userinfo.email",
+}
+
+func NotebooksInstanceScopesDiffSuppress(_, _, _ string, d *schema.ResourceData) bool {
```
Comment (SarahFrench) [inline mmv1/templates/terraform/constants/notebooks_instance.go:6]: I don't fully understand the problem with scopes. On the linked issue it was [noted that the API doesn't return service account scopes](https://github.com/hashicorp/terraform-provider-google/issues/16587#issuecomment-1822871108), but in the debug logs for [this failing test](https://github.com/GoogleCloudPlatform/magic-modules/pull/9915#issuecomment-1927798059) I can see `serviceAccountScopes` is present in responses from the API. I saw that there are two scopes that are [automatically used](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/notebooks_instance#service_account_scopes) when scopes aren't specified in the config by a user, so I can see why there may be a need for a diff suppress.

Could you please explain what scenarios the diff suppress would be needed in?

**Review by SarahFrench — COMMENTED** [review]
Some refactoring suggestions and a question about the issue with scopes.

Also, the latest test failure is due to a diff:

```
              ~ kms_key                = "projects/ci-test-project-188019/locations/global/keyRings/tf-test-tftest-shared-keyring-12bfr7fw32x/cryptoKeys/tf-test-tftest-shared-key-12bfr7fw32x/cryptoKeyVersions/1" -> "projects/ci-test-project-188019/locations/global/keyRings/tf-test-tftest-shared-keyring-12bfr7fw32x/cryptoKeys/tf-test-tftest-shared-key-12bfr7fw32x" # forces replacement
```

The POST request to create the resource doesn't include `cryptoKeyVersions/1` in the value, but the response from the API includes `cryptoKeyVersions/1`. This could be addressed by a diff suppress function also.

File: mmv1/templates/terraform/constants/notebooks_instance.go (hunk 1 above)
Comment (bcreddy-gcp) [reply mmv1/templates/terraform/constants/notebooks_instance.go:6]: The issue with scopes was that it was not returned in the response. We changed that in the backend to return the scopes. The issue currently is that `default_from_api: true` was not enabled and the 2 default scopes need to be always added for the instance to work, which is why this diff suppress.

**Comment by bcreddy-gcp** [comment]
Hey @SarahFrench, could you please take a look at the PR.

**Review by SarahFrench — APPROVED** [review]
Sorry for the delay - I was jiggling a few things and wanted to do a quick manual test of this PR before approving

**Pull request merged**